lib/progress.ts

```diff
@@ -129,16 +129,46 @@ export const getLastUnlockedLessonPath = (userProgressKey: string): string => {
   return path
 }
 
-export const getNextLessonKey = (userProgressKey: string): string => {
+export const getCurrentLessonKey = (
+  userProgressKey: string,
+  account?: any
+): string => {
+  if (!account) {
+    return userProgressKey
+  }
+
   const id = keys.indexOf(userProgressKey)
-  const result = keys[id + 1]
+  const nextChapterIntro = 'CH' + getNextChapterNumber(userProgressKey) + 'INT1'
 
-  if (!result) {
+  if (keys[id + 1] && keys[id + 1] === nextChapterIntro) {
```

Comment:
I'm not sure why on this line and on line 157 and I guess 161/162 we are adding either `1` or `2` to the `id`. Could you, perhaps with some comments, explain what it is you're doing here?

Reply:
Addressed in 1b95577a705b2b9b43bde705c3454ffa009caf27
Thanks!